lib/sync-DataSetModel.js

```diff
@@ -349,6 +355,14 @@ var self = {
     syncUtil.doLog(datasetClient.datasetId, 'silly', 'Now there are ' + datasetClient.pendingCallbacks.length + ' calbacks for datasetclient id =' + datasetClient.id, datasetClient);
   },
 
+  asyncGetClientHash: function(queryParams, metaData, cb) {
+    return syncUtil.asyncGenerateHash(queryParams, function(err, queryParamsHash) {
```

Comment:
@wei-lee - 
Do you think we need do do hashing twice for both meta and query? Do we have to know if any of them was changed separately?